docs/app/Examples/collections/Message/Types/MessageIconExample.js

```diff
@@ -1,18 +1,14 @@
-import React, { Component } from 'react'
-import { Message } from 'stardust'
+import React from 'react'
+import { Message, Icon } from 'stardust'
 
-export default class MessageIconExample extends Component {
-  render() {
-    return (
-      <div>
-        <Message icon='inbox' header='Have you heard about our mailing list?'>
-          Get the best news in your e-mail every day.
-        </Message>
+const MessageIconExample = () => (
+  <Message>
+    <Icon name='notched circle' loading />
+    <Message.Content>
+      <Message.Header>Just one second</Message.Header>
+      We are fetching that content for you.
+    </Message.Content>
+  </Message>
```

Comment:
Icon messages require an `icon` class.  We look for an `<Icon />` child and add this class.  I'm not 100% pleased with this as using another icon element or decorated `<Icon />` will not work.  The alternative is to require the icon prop:

```jsx
<Message icon>
  <Icon name='info' />
<Message>
```

This is also odd.  It is awkward in light of the intended usage of the icon prop:

```jsx
<Message icon='info' />
```

In this use case we add the icon className.  Since the prop API for markup is mutually exclusive children, allowing both a bool `icon` prop and children seems like something we shouldn't do.

### TL;DR

I'm not 100% happy with how we're adding the icon class, but I think it is the least of the evils.  Feedback welcome.